File: ReSharper.FSharp/src/FSharp.Psi/src/CodeFormatter/ZoneMarker.cs
using JetBrains.Application.BuildScript.Application.Zones;
using JetBrains.Platform.RdFramework;

namespace JetBrains.ReSharper.Plugins.FSharp.Psi.CodeFormatter
{
  [ZoneMarker]
  public class ZoneMarker : IRequire<IRdFrameworkZone>
  {
  }
}

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/CodeCompletion/ZoneMarker.fs
namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.CodeCompletion

open JetBrains.Application.BuildScript.Application.Zones
open JetBrains.Platform.RdFramework
open JetBrains.ProjectModel.NuGet
open JetBrains.ProjectModel.ProjectsHost.SolutionHost

[<ZoneMarker>]
type ZoneMarker() =
    interface IRequire<IHostSolutionZone>
    interface IRequire<INuGetZone>
    interface IRequire<IRdFrameworkZone>
